capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -316,4 +319,114 @@ else if (("remove_"+field.getName()).equals(key)) {
     }
     return getById(id, collectionReference, request, response, genericClass).get(0);
   }
-}
\ No newline at end of file
+
+  /**
+   * This method assures that the given post request is valid, contains the necessary fields, and 
+   * does not contain keys that do not match object fields exactly.
+   * @param {jsonObject} the supplied request body in json form
+   * @param {response} response returned from the call
+   * @param {fields} the Fields (contains name and type) of the class being used  
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return boolean true if it is a valid post request, false if not
+   */
+  public static <T extends BaseEntity> boolean postErrorHandler(JsonObject jsonObject, HttpServletResponse response, 
+      Field[] fields, List<String> requiredFields) throws IOException {
+    List<String> list = new ArrayList<>();
+    List<String> fieldNames = fields.stream().map(f -> f.getName()).collect(Collectors.toList());
+    list.addAll(jsonObject.keySet());
+    list.retainAll(fieldNames);
+    if (list.size() != jsonObject.keySet().size()) {
+      System.err.println("Error: Not all parameters in the request body are fields of the given class.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    list.clear();
+    list.addAll(jsonObject.keySet().stream()
+                    .filter(key -> jsonObject.get(key).getAsString().length() > 0)
+                    .collect(Collectors.toList()));
+    list.retainAll(requiredFields);
+    if (list.size() != requiredFields.size()) {
+      System.err.println("Error: Not all required fields are present in the request body, or some required fields are empty.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    return true;
+  }
```

Comment:
I think this could use a couple comments around each section of code in this. I know your System.err.println messages roughly indicate what it's supposed to do but it could be written out a bit more

Reply:
Sure I'll add.